presto-geospatial/src/main/java/com/facebook/presto/plugin/geospatial/GeoFunctions.java

```diff
@@ -971,26 +888,50 @@ public static Slice stEnvelope(@SqlType(GEOMETRY_TYPE_NAME) Slice input)
     public static Block stEnvelopeAsPts(@SqlType(GEOMETRY_TYPE_NAME) Slice input)
     {
         Envelope envelope = deserializeEnvelope(input);
-        if (envelope == null) {
+        if (envelope.isEmpty()) {
             return null;
         }
         BlockBuilder blockBuilder = GEOMETRY.createBlockBuilder(null, 2);
-        Point lowerLeftCorner = new Point(envelope.getXMin(), envelope.getYMin());
-        Point upperRightCorner = new Point(envelope.getXMax(), envelope.getYMax());
-        GEOMETRY.writeSlice(blockBuilder, serialize(createFromEsriGeometry(lowerLeftCorner, null, false)));
-        GEOMETRY.writeSlice(blockBuilder, serialize(createFromEsriGeometry(upperRightCorner, null, false)));
+        org.locationtech.jts.geom.Point lowerLeftCorner = createJtsPoint(envelope.getXMin(), envelope.getYMin());
+        org.locationtech.jts.geom.Point upperRightCorner = createJtsPoint(envelope.getXMax(), envelope.getYMax());
+        GEOMETRY.writeSlice(blockBuilder, serialize(lowerLeftCorner));
+        GEOMETRY.writeSlice(blockBuilder, serialize(upperRightCorner));
         return blockBuilder.build();
     }
 
+    @Description("Returns the bounding rectangle of a Geometry expanded by distance.")
+    @ScalarFunction("expand_envelope")
+    @SqlType(GEOMETRY_TYPE_NAME)
+    public static Slice expandEnvelope(@SqlType(GEOMETRY_TYPE_NAME) Slice input, @SqlType(DOUBLE) double distance)
+    {
+        if (isNaN(distance)) {
+            throw new PrestoException(INVALID_FUNCTION_ARGUMENT, "expand_envelope: distance is NaN");
+        }
+
+        if (distance < 0) {
+            throw new PrestoException(INVALID_FUNCTION_ARGUMENT, format("expand_envelope: distance %s is negative", distance));
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This code uses &#39;%s&#39; to format double: distance (declared at line 905). This is a potential locale-sensitive handling issue. It might cause errors in the handling and processing of the statement at line: 912. Consider formatting this data with &#39;%f&#39; instead.